slides/kube/logs-cli.md

```diff
@@ -0,0 +1,125 @@
+# Accessing logs from the CLI
+
+- The `kubectl logs` commands has limitations:
+
+  - it cannot stream logs from multiple pods at a time
+
+  - when showing logs from multiple pods, it mixes them all
+
+- We are going to see how to do better
+
+---
+
+## Doing it manually
+
+- We *could* (if we were so inclined), write a program or script that would:
+
+  - take a selector as an argument
+
+  - enumerate all pods matching that selector (`with kubectl get -l ...`)
+
+  - fork one `kubectl logs --follow ...` command per container
+
+  - annotate the logs (the output of each `kubectl logs ...` process) with their origin
+
+  - preserve ordering by using `kubectl logs --timestamps ...` and merge the output
+
+--
+
+- We *could* do it, but thankfully, others did it for us already!
+
+---
+
+## Stern
+
+[Stern](https://github.com/wercker/stern) is an open source project
+by [Wercker](http://www.wercker.com/).
+
+From the README:
+
+*Stern allows you to tail multiple pods on Kubernetes and multiple containers within the pod. Each result is color coded for quicker debugging.*
+
+*The query is a regular expression so the pod name can easily be filtered and you don't need to specify the exact id (for instance omitting the deployment id). If a pod is deleted it gets removed from tail and if a new is added it automatically gets tailed.*
+
+Exactly what we need!
+
+---
+
+## Installing Stern
+
+- For simplicity, let's just grab a binary release
+
+.exercise[
+
+- Download a binary release from GitHub:
+  ```bash
+  sudo curl -L -o /usr/local/bin/stern \
+       https://github.com/wercker/stern/releases/download/1.6.0/stern_linux_amd64
+  sudo chmod +x /usr/local/bin/stern
+  ```
+
+]
+
+(The installation will differ if you want to run Stern on anything else than a Linux amd64 machine.)
```

Comment:
Oh, this is executed on the cluster — which runs Linux in that case.

Do you think I should clarify?

Reply:
Oh, right, of course! I think it was confusing to me because a lot of add-on tools are something people might try to use locally, not on their cluster. I think in this case it's fine, but maybe we want to change:

 > (The installation will differ if you want to run Stern on anything else than a Linux amd64 machine.)

to say something like "(The installation will differ if you want to run Stern on anything else than a Linux amd64 machine, as this workshop uses.)" ? People might not be thinking of "nodeN" as "Linux amd64" specifically.

Reply:
Right on! Let me know if the new wording seems better.